src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -558,8 +563,28 @@ OverlayImpl::onPrepare()
                 std::string const& name,
                 std::vector <beast::IP::Endpoint> const& addresses)
             {
-                if (!addresses.empty ())
-                    m_peerFinder->addFixedPeer (name, addresses);
+                std::vector <beast::IP::Endpoint> addresses_update;
+                addresses_update.reserve(addresses.size());
+                for (auto const& addr : addresses)
+                {
+                    if (addr.port () == 0)
```

Comment:
Same as above. I also don't quite like the introduction of a temporary here. Thought: remove `const&` from the lambda argument list (so `addresses` is passed by copy, potentially leveraging move semantics) and then:

    for(auto& addr : addresses)
    {
        if (addr.port() == 0)
            addr = addr.at_port(51235);
    }

Reply:
This requires change to the `Resolve` class/subclasses to remove the `const`; i.e. semantics changes. Current implementation doesn't require the change at the expense of copying. But this only happens at start up and most likely the number of fixed peers is not substantial.